src/helix-ui/styles/components/checkbox.less

```diff
@@ -0,0 +1,255 @@
+  -ms-grid-column-align: start;
+  display: -ms-grid;
+  display: grid;
+  grid-template-rows: auto auto;
+  grid-template-columns: auto 1fr;
+  grid-template-areas:
+    'ctrl lbl'
+    '.    help';
+  justify-items: start;
+
+  // move below eventually
+  > label {
+    // grid child
+    -ms-grid-column: 2;
+    -ms-grid-row: 1;
+    grid-area: lbl;
+
+    // grid parent
+    -ms-grid-columns: auto 1fr;
+    -ms-grid-rows: auto;
+    //align-items: baseline;
+    display: -ms-grid;
+    display: grid;
+    grid-template-areas: 'facade txt';
+    grid-template-columns: auto 1fr;
+    grid-template-rows: auto;
+    grid-gap: 0.5rem;
+
+    // other
+    margin: 0;
+    padding: 0;
+    //user-select: none;
+
+    > hx-checkbox {
+      -ms-grid-column: 1;
+      -ms-grid-row: 1;
+      grid-area: facade;
+      display: none;
+      margin-right: 0.50rem;
+    }
+
+    > span {
+      -ms-grid-column: 2;
+      -ms-grid-row: 1;
+      grid-area: txt;
+      line-height: 1.5;
+    }
+  }
+
+  // help text
+  > p {
+    -ms-grid-column: 2;
+    -ms-grid-row: 2;
+    font-size: 0.75rem;
+    grid-area: help;
+    margin: 0;
+    order: 2;
+  }
+
+  > input[type="checkbox"] {
+    background-color: transparent;
+    color: @gray-500;
+    display: inline-block;
+    height: 1rem;
+    width: 1rem;
+
+    &::-ms-check {
+      background-color: @gray-0;
+      border-color: currentColor;
+      border-style: solid;
+      border-width: 1px;
+      box-sizing: border-box;
+      color: inherit;
+      display: block;
+    }
+
+    &:checked,
+    &:indeterminate {
+      color: @cyan-900;
+    }
+
+    &:focus {
+      box-shadow: @focus-glow;
+    }
+
+    &:hover {
+      color: @cyan-500;
+
+      &::-ms-check {
+        background-color: @cyan-50;
+        border-color: @cyan-500;
+      }
+    }
+
+    &:invalid {
+      color: @red-900;
+
+      &::-ms-check {
+        border-width: 2px;
+      }
+
+      &:focus {
+        box-shadow: @focus-glow-invalid;
+
+        &::-ms-check {
+          border-color: @red-900;
+          color: @red-900;
+        }
+      }
+
+      &:hover {
+        &::-ms-check {
+          background-color: @red-100;
+          border-color: @red-500;
+          color: @red-900;
+        }
+      }
+    }
+
+    &:disabled {
+      color: @gray-500;
+      cursor: not-allowed;
+
+      &::-ms-check {
+        background-color: @gray-100 !important;
+        border-width: 1px;
+      }
+
+      &:focus {
+        &::-ms-check {
+          box-shadow: @focus-glow-disabled;
+        }
+      }
+
+      &:hover {
+        &::-ms-check {
+          background-color: @gray-100;
+          border-color: @gray-500;
+          color: @gray-500;
+        }
+      }
+    }
+  }
+}
+
+
+@supports (display: grid) {
+  hx-checkbox-control {
+    > label > hx-checkbox {
+      margin-right: 0;
+    }
+  }
+}
+
+@supports (display: inline-grid) {
+  hx-checkbox-control {
+    > input[type="checkbox"] {
+      #Mix.hideInput();
```

Comment:
Let's refactor this to keep as much of the state styling together (outside of the `@supports`). Styles in the `@supports` should be very minimal.